### PR TITLE
Bug 1335092 - Improve UX

### PR DESCRIPTION
**Pull request by jezdez**
screen shot 2017-01-30 at 18 09 42-fullpage

----

screen shot 2017-01-30 at 18 07 58-fullpage

### CODE REVIEW

File: atmo/clusters/forms.py [hunk 1]
```diff
@@ -10,18 +10,12 @@
...
 class NewClusterForm(FormControlFormMixin, CreatedByModelFormMixin,
                      forms.ModelForm):
     prefix = 'new'
 
     identifier = forms.RegexField(
-        label='Cluster identifier',
+        label='Identifier',
         required=True,
         regex=r'^[\w-]{1,100}$',
         widget=forms.TextInput(attrs={
```
Comment (robhudson) [inline atmo/clusters/forms.py]: should we make the Spark cluster identifier help text match the job identifier help text?

File: atmo/clusters/forms.py [hunk 2]
```diff
@@ -56,12 +50,28 @@ class NewClusterForm(FormControlFormMixin, CreatedByModelFormMixin,
     class Meta:
         model = models.Cluster
         fields = ['identifier', 'size', 'ssh_key', 'emr_release']
+        widgets = {
+            'emr_release': forms.RadioSelect(attrs={
+                'required': 'required',
+                'class': 'radioset',
+            }),
+        }
```
Comment (robhudson) [inline atmo/clusters/forms.py]: I like this. 👍

File: atmo/jobs/forms.py [hunk 3]
```diff
@@ -97,28 +99,28 @@ class BaseSparkJobForm(FormControlFormMixin, CachedFileModelFormMixin,
...
                   'not be disabled.',
     )
     notebook = CachedFileField(
         required=True,
-        widget=forms.FileInput(),  # no need to specific required attr here
+        widget=forms.FileInput(attrs={'accept': '.ipynb'}),
         label='Analysis Jupyter Notebook',
-        help_text='A Jupyter/IPython Notebook with a file ipynb '
+        help_text='A Jupyter/IPython Notebook with a file .ipynb '
                   'extension.'
```
Comment (robhudson) [inline atmo/jobs/forms.py]: This reads better to me as "... with a .ipynb file extension."

File: atmo/jobs/forms.py [hunk 4]
```diff
@@ -168,21 +170,27 @@ class NewSparkJobForm(BaseSparkJobForm):
...
                   'visible in the AWS management console.'
     )
 
     notebook = CachedFileField(
         required=False,
-        widget=forms.FileInput(),
+        widget=forms.FileInput(attrs={'accept': '.ipynb'}),
         label='Analysis Jupyter Notebook',
         help_text='Optional Jupyter/IPython Notebook with a file ipynb '
                   'extension.'
```
Comment (robhudson) [inline atmo/jobs/forms.py]: We should update the help text here to match also.

File: atmo/templates/atmo/_form.html [hunk 5]
```diff
@@ -8,38 +8,53 @@
...
-              <span class="file-label hidden">Browse&hellip;</span>
-              {{ field }}
-          </span>
+    <label class="input-group-btn">
+      <span class="btn btn-default">
+        <span class="file-label hidden">Browse&hellip;</span>
+        {{ field }}
+        </span>
       </label>
       <input type="text" class="form-control hidden file-selection" readonly>
```
Comment (robhudson) [inline atmo/templates/atmo/_form.html]: The indenting is off here.

File: atmo/static/js/keys.js [hunk 6]
```diff
@@ -14,9 +14,16 @@ $(function() {
       var reader = new FileReader();
       // Closure to capture the file information.
       reader.onload = function(event) {
-        if (!title_input.val()) {
-          title_input.val(file.name.replace(/.pub/g, ''));
-        };
+        // use the filename..
+        var title = file.name.replace(/.pub/g, '');
+        // or the -- if available -- the user found in the key content
```
Comment (maurodoglio) [inline atmo/static/js/keys.js]: the second `the` should probably be removed

File: atmo/clusters/forms.py (hunk 1 above)
Comment (jezdez) [inline atmo/clusters/forms.py]: Yeah, good idea.

File: atmo/jobs/forms.py (hunk 3 above)
Comment (jezdez) [inline atmo/jobs/forms.py]: Makes sense, will change.

**Pull request merged**